tests/hipscat_import/association/test_association_argument.py

```diff
@@ -0,0 +1,142 @@
+        AssociationArguments(
+            primary_input_catalog_path=None,  ## empty
+            primary_id_column="id",
+            primary_join_column="id",
+            join_input_catalog_path=small_sky_object_catalog,
+            join_id_column="id",
+            join_foreign_key="id",
+            output_path=tmp_path,
+            output_catalog_name="small_sky_self_join",
+        )
+
+    with pytest.raises(ValueError, match="primary_id_column"):
+        AssociationArguments(
+            primary_input_catalog_path=small_sky_object_catalog,
+            primary_id_column="",  ## empty
+            primary_join_column="id",
+            join_input_catalog_path=small_sky_object_catalog,
+            join_id_column="id",
+            join_foreign_key="id",
+            output_path=tmp_path,
+            output_catalog_name="small_sky_self_join",
+            overwrite=True,
+        )
+
+    with pytest.raises(ValueError, match="primary_join_column"):
+        AssociationArguments(
+            primary_input_catalog_path=small_sky_object_catalog,
+            primary_id_column="id",
+            primary_join_column="",  ## empty
+            join_input_catalog_path=small_sky_object_catalog,
+            join_id_column="id",
+            join_foreign_key="id",
+            output_path=tmp_path,
+            output_catalog_name="small_sky_self_join",
+            overwrite=True,
+        )
+
+    with pytest.raises(ValueError, match="join_input_catalog_path"):
+        AssociationArguments(
+            primary_input_catalog_path=small_sky_object_catalog,
+            primary_id_column="id",
+            primary_join_column="id",
+            join_input_catalog_path="",  ## empty
+            join_id_column="id",
+            join_foreign_key="id",
+            output_path=tmp_path,
+            output_catalog_name="small_sky_self_join",
+            overwrite=True,
+        )
+
+    with pytest.raises(ValueError, match="join_id_column"):
+        AssociationArguments(
+            primary_input_catalog_path=small_sky_object_catalog,
+            primary_id_column="id",
+            primary_join_column="id",
+            join_input_catalog_path=small_sky_object_catalog,
+            join_id_column="",  ## empty
+            join_foreign_key="id",
+            output_path=tmp_path,
+            output_catalog_name="small_sky_self_join",
+            overwrite=True,
+        )
+
+    with pytest.raises(ValueError, match="join_foreign_key"):
+        AssociationArguments(
+            primary_input_catalog_path=small_sky_object_catalog,
+            primary_id_column="id",
+            primary_join_column="id",
+            join_input_catalog_path=small_sky_object_catalog,
+            join_id_column="id",
+            join_foreign_key="",  ## empty
+            output_path=tmp_path,
+            output_catalog_name="small_sky_self_join",
+            overwrite=True,
+        )
+
```

Comment:
Are the `output_path` and `output_catalog_name` not required?

Reply:
Yeah, they are, but I don't **have** to test it here to get coverage credit for it.

Thanks for keeping me honest.